nixos/modules/config/nix.nix

```diff
@@ -386,5 +386,11 @@ in
         )
       );
     };
+
+    # Enable git if flake support is enabled
+    programs.git = mkIf (builtins.match ".*flakes.*" != null) {
```

Comment:
I don't know how this work. This is a syntax error.

Reply:
Howso?

Reply:
Here only regex, no str for.

Reply:
Also we don't need a regex here. We should match if any entry of nix.settings.experimental_features is flakes.

Reply:
`nix.settings.experimental_features` isn't a thing though???

https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=experimental_features